app/src/main/java/com/keylesspalace/tusky/view/MediaPreviewLayout.kt

```diff
@@ -181,9 +185,13 @@ class MediaPreviewLayout(context: Context, attrs: AttributeSet? = null) :
         }
     }
 
-    inline fun forEachIndexed(action: (Int, MediaPreviewImageView) -> Unit) {
+    inline fun forEachIndexed(action: (Int, MediaPreviewImageView, TextView) -> Unit) {
         for (index in 0 until childCount) {
-            action(index, getChildAt(index) as MediaPreviewImageView)
+            action(
+                index,
+                getChildAt(index).findViewById(R.id.preview_image_view) as MediaPreviewImageView,
```

Comment:
Yeah I wrote this `forEachIndexed` purely for convenience sake. I do worry it has a bit too much knowledge of the layout's details and maybe it would be better to do the looping/indexing outside of this class instead? I think it's only called from one place anyway.

Reply:
The part in the Layout class is probably alright.

It enables however much manipulation of class knowledge elsewhere.

Do you have an idea on how to improve it?

Reply:
Have been thinking about this a bit and it seems to me that `setMediaPreviews` (and `loadImage` and `sensitiveMediaWarning`) should move fully to `MediaPreviewLayout`?

Then only the current data from the current StatusBaseViewHolder.setMediaPreviews must be handed over and this is ok (data models and listeners).

But that would be a medium move (I failed on a ten minute attempt just now).

Reply:
Depends on what we think should be the responsibility of this class. I was originally going to have it encompass most of the work for displaying previews, but I realized that would be a significant change to the existing code so I moved it in the opposite direction, only focusing on layout and nothing more, which in hindsight I don't think is a bad thing because it separates the application logic from the reasonably complex layout code. This could actually be even more generic by passing the the layout id for it's children so it would have no knowledge of what the child views actually contain, but not convinced that level of abstraction is really necessary so eh, what you are currently doing seems fine to me.

> Do you have an idea on how to improve it?

You could try using viewbinding here, ex: hold an array of `ItemImagePreviewOverlayBinding`, would save you some findViewById calls.